axum-core/src/response/mod.rs

```diff
@@ -17,3 +17,52 @@ pub use self::{
 /// Type alias for [`http::Response`] whose body type defaults to [`BoxBody`], the most common body
 /// type used with axum.
 pub type Response<T = BoxBody> = http::Response<T>;
+
+/// A flexible [IntoResponse]-based result type
+///
+/// All types which implement [IntoResponse] can be converted to an [Error].
+/// This makes it useful as a general error type for functions which combine
+/// multiple distinct error types but all of which implement [IntoResponse].
+///
+/// For example, note that the error types below differ. However, both can be
+/// used with the [Result], and therefore the `?` operator, since they both
+/// implement [IntoResponse].
+///
+/// ```no_run
+/// use axum_core::response::{IntoResponse, Response, Result};
+/// use http::StatusCode;
+///
+/// fn foo() -> Result<&'static str> {
+///     Err((StatusCode::NOT_FOUND, "not found"))?;
+///     Err(StatusCode::BAD_REQUEST)?;
+///     Ok("ok")
+/// }
```

Comment:
I think example should show something slightly more realistic. Perhaps calling a few other fallible functions with different error types. I don't expect users actually do `Err((StatusCode::NOT_FOUND, "not found"))?` in their own code 😅

Reply:
See the latest commit. Is this what you had in mind?